app/[locale]/academics/helper/courses/CourseEditor.tsx

```diff
@@ -0,0 +1,198 @@
+import { putCourseAction } from '@/actions/academics';
+import BookmarkIcon from '@/public/image/bookmark_icon.svg';
+
+import Dropdown from '@/components/common/form/Dropdown';
+import BasicTextInput from '@/components/editor/common/BasicTextInput';
+
+import { CLASSIFICATION, Course, GRADE } from '@/types/academics';
+import { getKeys } from '@/types/object';
+
+import { validateCourseForm } from '@/utils/formValidation';
+import { handleServerAction } from '@/utils/serverActionError';
+import { errorToast, successToast } from '@/utils/toast';
+
+import useCourseEditor from './useCourseEditor';
+
+const CREDIT = [1, 2, 3, 4];
+
+export default function CourseEditor({
+  initCourse,
+  toggleEditMode,
+  setCourse,
+}: {
+  initCourse: Course;
+  toggleEditMode: () => void;
+  setCourse: (course: Course) => void;
+}) {
+  const { content, setContentByKey, setLanguageContent, setClassification } =
+    useCourseEditor(initCourse);
+  const isGraduateCourse = initCourse.grade === 0;
```

Comment:
드롭다운쪽에서만 쓰이는거 같은데 컨텐츠 전체를 여기서 만들면 왜 필요한지 명확할 것 같아

```suggestion
  const dropdownContent = isGraduateCourse ? [GRADE[0]] : GRADE.slice(1);
```